docs/Modeling-Artifacts.md

```diff
@@ -0,0 +1,378 @@
+`sha256:9c7c6bfa51dac3c9dfeffc7a0a795c30101f1f60afa64739767cedd92f574570`.
+
+The relationship between the Index and the artifacts in the CAS can be modeled as the graph below:
+
+```mermaid
+graph TD;
+
+Index["Index<br>(sha256:9c7c6b...)"]--manifests-->Manifest
+Index--manifests-->AnotherManifest
+
+Manifest--layers-->Layer0["Layer blob 0<br>(sha256:b5bb9d...)"]
+Manifest--layers-->Layer1["Layer blob 1<br>(sha256:7d865e...)"]
+Manifest["Manifest<br>(sha256:314c7f...)"]--config-->Config["Config blob<br>(sha256:44136f...)"]
+
+AnotherManifest["Another Manifest<br>(sha256:eba50b)"]--config-->Config
+AnotherManifest--layers-->Layer2["Layer blob 2<br>(sha256:a94890...)"]
+
+```
+
+In this graph, the Index serves as the root of the overall graph, with each manifest defining the root of its corresponding artifact subgraph.
+
+## Graph Concepts
+
+A complex DAG may integrate artifacts, their referrers, and the Indexes that reference them. The following example demonstrates such a graph:
+
+```mermaid
+graph TD;
+
+I0["Index i0"]--manifests-->M0
+I0--manifests-->M1
+
+M2["Manifest m2"]--config-->Blob0
+M2--layers-->Blob5["Blob b5"]
+M2--subject-->M0
+
+M1["Manifest m1"]--config-->Blob3["Blob b3"]
+M1--layers-->Blob4["Blob b4"]
+
+M0["Manifest m0"]--config-->Blob0["Blob b0"]
+M0--layers-->Blob1["Blob b1"]
+M0--layers-->Blob2["Blob b2"]
+```
+
+For any node in the graph, the following definitions apply:
+
+- **Successor:** Any node that is pointed to by a given node. For instance:
+  - Blob `b0` is a successor of both `m0` and `m2`
+  - Manifest `m0` is a successor of `m2` and `i0`
+  
+- **Predecessor:** Any non-leaf node that directly points to a given node. For instance:
+  - Manifest `m0` is a predecessor of `b0`, `b1`, and `b2`
+  - Manifest `m2` is a predecessor of `b0`, `b5`, and `m0`
+  - Index `i0` is a predecessor of `m0` and `m1`
+
+These definitions apply to nodes of any type—manifests, indexes, or arbitrary blobs.
+
+However, the referrer relationship is different. A manifest (including [Image Manifest](https://github.com/opencontainers/image-spec/blob/v1.1.1/manifest.md) and [Index](https://github.com/opencontainers/image-spec/blob/v1.1.1/image-index.md)) with a `subject` field is considered a referrer of that subject manifest. According to [OCI image-spec v1.1.1](https://github.com/opencontainers/image-spec/blob/v1.1.1/manifest.md), both the `referrer` and the `subject` must be manifests.
+
+So, it is worth noting that:
+
+- `m0` is a `subject` of `m2`, and it is a successor of both `m2` and `i0`
+- `m2` is a referrer of `m0`, and it is a predecessor of `m0`, `b0`, and `b5`
+
+Defining functions `Predecessors()`, `Successors()`, and `Referrers()`, the example result would be:
+
+```
+Successors(m0) == [b0, b1, b2]
+Predecessors(m0) == [m2, i0]
+Referrers(m0) == [m2]
+
+Successors(m2) == [m0, b0, b5]
+Predecessors(m2) == []
+Referrers(m2) == []
+
+Successors(b0) == []
+Predecessors(b0) == [m0, m2]
+Referrers(b0) == []
```

Comment:
nit: Just thinking if this renders better
```suggestion
  Successors(m0) == [b0, b1, b2]
Predecessors(m0) == [m2, i0]
   Referrers(m0) == [m2]

  Successors(m2) == [m0, b0, b5]
Predecessors(m2) == []
   Referrers(m2) == []

  Successors(b0) == []
Predecessors(b0) == [m0, m2]
   Referrers(b0) == []
```

Reply:
Looks nice😀